spotlight/losses.py

```diff
@@ -188,10 +203,12 @@ def regression_loss(observed_ratings, predicted_ratings):
 
     assert_no_grad(observed_ratings)
 
-    return ((observed_ratings - predicted_ratings) ** 2).mean()
+    return (
+        ((observed_ratings - predicted_ratings) * weights) ** 2
```

Comment:
I don't think we should be squaring the weights: compute the squared error, _then_ weight.